orchestration/src/main/java/com/sap/ai/sdk/orchestration/spring/OrchestrationChatOptions.java

```diff
@@ -0,0 +1,224 @@
+package com.sap.ai.sdk.orchestration.spring;
+
+import static com.sap.ai.sdk.orchestration.OrchestrationAiModel.Parameter.FREQUENCY_PENALTY;
+import static com.sap.ai.sdk.orchestration.OrchestrationAiModel.Parameter.MAX_TOKENS;
+import static com.sap.ai.sdk.orchestration.OrchestrationAiModel.Parameter.PRESENCE_PENALTY;
+import static com.sap.ai.sdk.orchestration.OrchestrationAiModel.Parameter.TEMPERATURE;
+import static com.sap.ai.sdk.orchestration.OrchestrationAiModel.Parameter.TOP_P;
+
+import com.sap.ai.sdk.orchestration.OrchestrationModuleConfig;
+import com.sap.ai.sdk.orchestration.model.LLMModuleConfig;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.val;
+import org.springframework.ai.chat.prompt.ChatOptions;
+
+/** Configuration to be used for orchestration requests. */
+@Data
+@Getter(AccessLevel.NONE)
+@Setter(AccessLevel.NONE)
+public class OrchestrationChatOptions implements ChatOptions {
+
+  @Getter(AccessLevel.PUBLIC)
+  @Setter(AccessLevel.PUBLIC)
+  @Nonnull
+  OrchestrationModuleConfig config;
+
+  /**
+   * Returns the model to use for the chat.
+   *
+   * @return the model to use for the chat
+   * @see com.sap.ai.sdk.orchestration.OrchestrationAiModel
+   */
+  @Nonnull
+  @Override
+  public String getModel() {
+    return getLlmConfigNonNull().getModelName();
+  }
+
+  private void setModel(@Nonnull final String model) {
+    getLlmConfigNonNull().setModelName(model);
+  }
+
+  /**
+   * Returns the model version to use for the chat. "latest" by default.
+   *
+   * @return the model version to use for the chat.
+   */
+  @Nonnull
+  public String getModelVersion() {
+    return getLlmConfigNonNull().getModelVersion();
+  }
+
+  private void setModelVersion(@Nonnull final String modelVersion) {
+    getLlmConfigNonNull().setModelVersion(modelVersion);
+  }
```

Comment:
(Minor)

Do we really need these private setter one-liners? :)

Reply:
removed